### PR TITLE
[extension/httpforwarder] rename httpforwarder to httpforwarderextension

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Deprecate the httpforwarder extension
Introduce the httpforwarderextension extension.

**Link to tracking Issue:**
Relates to #24171

### CODE REVIEW

File: .chloggen/rename_httpforwarder_extension.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: breaking
```
Comment (astencel-sumo) [inline .chloggen/rename_httpforwarder_extension.yaml:7]: Component: breaking? 🤔  Shouldn't it be:
```suggestion
component: httpforwarder
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -94,6 +94,7 @@ body:
       - extension/headerssetter
       - extension/healthcheck
       - extension/httpforwarder
+      - extension/httpforwarder
```
Comment (astencel-sumo) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:97]: Is this duplicate line required? 😅

File: reports/distributions/contrib.yaml [hunk 3]
```diff
@@ -67,6 +67,7 @@ components:
         - health_check
         - host_observer
         - http_forwarder
+        - http_forwarder
```
Comment (astencel-sumo) [inline reports/distributions/contrib.yaml:70]: I don't think Contrib distro will include two `http_forwarder` extensions?

Comment (astencel-sumo) [reply reports/distributions/contrib.yaml:70]: Oh these reports are generated from the metadata. I wonder if metadata should be updated?

File: extension/httpforwarderextension/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+type: http_forwarder
+
+status:
+  class: extension
+  stability:
+    beta: [extension]
+  distributions: [contrib, splunk, sumo]
```
Comment (astencel-sumo) [inline extension/httpforwarderextension/metadata.yaml:7]: This causes the distribution reports to include the `http_forwarder` extension twice. How about we make this list empty and only update when the distros actually switch to the new component?
```suggestion
  distributions: []
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (atoulme) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:97]: It is :/

File: reports/distributions/contrib.yaml (hunk 3 above)
Comment (atoulme) [reply reports/distributions/contrib.yaml:70]: no I don't want to change the type of the component.

File: extension/httpforwarderextension/metadata.yaml (hunk 4 above)
Comment (atoulme) [reply extension/httpforwarderextension/metadata.yaml:7]: Got it. I will also deprecate the old extension.

File: extension/httpforwarder/go.mod [hunk 5]
```diff
@@ -1,3 +1,4 @@
+// Deprecated: use httpforwarderextension instead
```
Comment (astencel-sumo) [inline extension/httpforwarder/go.mod:1]: This is good. Ideally we should also include this in the README?

**Review by andrzej-stencel — APPROVED** [review]

File: .chloggen/rename_httpforwarder_extension.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (dmitryax) [inline .chloggen/rename_httpforwarder_extension.yaml:27]: THis should be API only change, right? Users should not be affected.

Comment (atoulme) [reply .chloggen/rename_httpforwarder_extension.yaml:27]:
```suggestion
change_logs: [api]
```

File: extension/httpforwarderextension/go.mod [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../../internal/common
+
+retract (
+	v0.76.2
+	v0.76.1
+	v0.65.0
+)
```
Comment (dmitryax) [inline extension/httpforwarderextension/go.mod:75]: Not needed

**Review by dmitryax — APPROVED** [review]

File: extension/httpforwarderextension/go.mod (hunk 7 above)
Comment (atoulme) [reply extension/httpforwarderextension/go.mod]: Done, removed.

**Comment by martin-majlis-s1** [comment]
This PR has broken bug reporting - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/ISSUE_TEMPLATE/bug_report.yaml as well as feature requests - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/ISSUE_TEMPLATE/feature_request.yaml


Screenshot 2024-02-17 at 6 23 25
Screenshot 2024-02-17 at 6 22 39

**Pull request merged**